src/app/shared/directives/infiniteScroll/infinite-scroll/scroll-direction.enum.ts

```diff
@@ -0,0 +1,6 @@
+enum ScrollDirection {
+    up,
+    down,
+}
+
+export default ScrollDirection;
```

Comment:
Почему не `export enum`?

Вообще, лучшей практикой считается указание для `enum` уникальное значение, это добавит безопасность на уровне проверок JS, плюс даст возможность настроить более прозрачное и менее дорогое логирование, если понадобится где-то
```suggestion
export enum ScrollDirection {
    up = 'up',
    down = 'down',
}
```

Reply:
Добавил(уникальные) ключи. 
Только не до конца понял почему нас значения по умолчанию не устраивают? Они же будут все ровно по значению сравниваться, а значения не повторяются.

Reply:
По умолчанию, мы говорим что больше ничего не собираемся добавлять в этот модуль.

Reply:
Самый яркий кейс: представь, что ты хочешь сделать декоративный механизм логирования, при помощи, например, декораторов. Сможем ли мы это легко сделать со значниями enum 0, 1, .... Кажется, что нет, ведь если в логах мы увидим что вызвалась та или иная функция с параметрами 0 или 1 или ..., то вряд ли мы что-то сразу поймем.

А потом, мы можем вспомнить NgRx, где обычно создают enum с типами событий, таких enum может быть несколько. Все события будут попадать в общий механизм для дальнейшей обработки и события с повторяющимеся типами будут восприниматься как одинаковые события, что сломает нам всю логику.

В общем, я бы эту практику свел к тому, что понятные значения лучше чем числа